semconv/v1.34.0/error_type_test.go

```diff
@@ -0,0 +1,66 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package semconv // import "go.opentelemetry.io/otel/semconv/v1.34.0"
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"testing"
+
+	"go.opentelemetry.io/otel/attribute"
+)
+
+type CustomError struct{}
+
+func (CustomError) Error() string {
+	return "custom error"
+}
+
+func TestErrorType(t *testing.T) {
+	customErr := CustomError{}
+	builtinErr := errors.New("something went wrong")
+	var nilErr error
+
+	wantCustomType := reflect.TypeOf(customErr)
+	wantCustomStr := fmt.Sprintf("%s.%s", wantCustomType.PkgPath(), wantCustomType.Name())
+
+	tests := []struct {
+		name    string
+		err     error
+		want    attribute.KeyValue
+		wantStr string
+	}{
+		{
+			name: "BuiltinError",
+			err:  builtinErr,
+			want: attribute.String("error.type", "*errors.errorString"),
+		},
+		{
+			name:    "CustomError",
+			err:     customErr,
+			wantStr: wantCustomStr,
+		},
+		{
+			name: "NilError",
+			err:  nilErr,
+			want: ErrorTypeOther,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := ErrorType(tt.err)
+			if tt.name == "CustomError" {
```

Comment:
This should not gate off of a test name. It either needs to be its own test, the testing struct type needs to be updated to make the field general enough to support all cases the same, or this testing scope needs to be generalized to handle the test case without relying on the test name.

Reply:
Correct. This is the reason behind https://github.com/open-telemetry/opentelemetry-go/pull/6962#discussion_r2189405036

Reply:
I tried to "flatten" the test now.